lib/optimizely/event_builder.rb

```diff
@@ -41,198 +42,202 @@ def ==(event)
-    end
+      decision = [{
+        campaign_id: @config.experiment_key_map[experiment_key]['layerId'],
+        experiment_id: experiment_id,
+        variation_id: variation_id,
+      }]
+
+      event = [{
+        entity_id: @config.experiment_key_map[experiment_key]['layerId'],
+        timestamp: get_timestamp(),
+        key: ACTIVATE_EVENT_KEY,
+        uuid: get_uuid()
+      }]
+
+      impressionEventParams = {
+        decisions: decision,
+        events: event
+      }
 
-    def add_user_id(user_id)
-      @params['visitorId'] = user_id
+      impressionEventParams;
     end
 
-    def add_attributes(attributes)
-      @params['userFeatures'] = []
-
-      return if attributes.nil?
+    def get_conversion_event_params(event_key, event_tags, experiment_variation_map)
+      # Creates object of params specific to conversion events
+      #
+      # event_key -                +String+ Event key representing the event which needs to be recorded
+      # event_tags -               +Hash+ Values associated with the event.
+      # experiment_variation_map - +Hash+ Map of experiment IDs to bucketed variation IDs
+      #
+      # Returns +Hash+ Impression event params
 
-      attributes.keys.each do |attribute_key|
-        # Omit falsy attribute values
-        attribute_value = attributes[attribute_key]
-        next unless attribute_value
+      conversionEventParams = []
 
-        # Skip attributes not in the datafile
-        attribute_id = @config.get_attribute_id(attribute_key)
-        next unless attribute_id
+      experiment_variation_map.each do |experiment_id, variation_id|
 
-        feature = {
-          'id' => attribute_id,
-          'name' => attribute_key,
-          'type' => 'custom',
-          'value' => attribute_value,
-          'shouldIndex' => true,
+        decision = {
+          decisions: [{
+            campaign_id: @config.experiment_id_map[experiment_id]['layerId'],
+            experiment_id: experiment_id,
+            variation_id: variation_id,
+          }],
+          events: [],
         }
-        @params['userFeatures'].push(feature)
-      end
-    end
-
-    def add_decision(experiment, variation_id)
-      experiment_key = experiment['key']
-      experiment_id = experiment['id']
-      @params['layerId'] = @config.experiment_key_map[experiment_key]['layerId']
-      @params['decision'] = {
-        'variationId' => variation_id,
-        'experimentId' => experiment_id,
-        'isLayerHoldback' => false,
-      }
-    end
 
-    def add_event_tags(event_tags)
-      @params['eventFeatures'] ||= []
-      @params['eventMetrics'] ||= []
+        event_object = {
+          entity_id: @config.event_key_map[event_key]['id'],
+          timestamp: get_timestamp(),
+          uuid: get_uuid(),
+          key: event_key,
+        }
 
-      return if event_tags.nil?
+        if event_tags
+          event_revenue = Helpers::EventTagUtils.get_revenue_value(event_tags)
```

Comment:
I find the variable names event_revenue and event_value confusing. It feels like the variables hold some event object. It would be more convenient if they are renamed to something like 'revenue_value' or just 'revenue'

Reply:
there is value is a property of event and not of revenue. Therefore revenue_value dosent make sense. both these field are part of event so event_revenue and event_value make more sense

in Node sdk value is stored in eventValue variable. and used uniform name for revent and value fields as event_revenue and event_value. https://github.com/optimizely/node-sdk/blob/94f940cff1872ec1023fd9eb2ad4a3781d31d61d/lib/core/event_builder/index.js#L131

I have mostly used node sdk for reference as it was explicitly requested in requirement document